File: easy/13. Roman to Integer/Solution.java
class Solution {
    private static final Map<Character, Integer> symbols;

    static {
        symbols = new HashMap<>();
        symbols.put('I', 1);
        symbols.put('V', 5);
        symbols.put('X', 10);
        symbols.put('L', 50);
        symbols.put('C', 100);
        symbols.put('D', 500);
        symbols.put('M', 1000);
    }

    public int romanToInt(String s) {
        int sum = 0;
        int length = s.length();
        for(int i = 0; i < length; i++){
            int value = symbols.get(s.charAt(i));
            if(i != length - 1) {
                int nextValue = symbols.get(s.charAt(i + 1));
                if(nextValue > value){
                    sum -= value;
                    continue;
                }
            }
            sum += value;
        }
        return sum;
    }
}

File: easy/13. Roman to Integer/Solution.kt
class Solution {
    private object Symbols {
        val map: Map<Char, Int>  = mapOf(
            'I' to 1, 'V' to 5, 'X' to 10,
            'L' to 50, 'C' to 100, 'D' to 500, 'M' to 1000
        )
    }

    fun romanToInt(s: String): Int {
        var sum = 0
        val map: Map<Char, Int> = Symbols.map;
        s.forEachIndexed{
            i, c ->
            val value = map[c]!!
            if(i < s.lastIndex && value < map[s[i + 1]]!!){
                sum -= value
            } else {
                sum += value
            }
        }
        return sum;
    }
}